src/hotspot/share/opto/matcher.cpp

```diff
@@ -965,6 +988,12 @@ void Matcher::init_spill_mask( Node *ret ) {
   idealreg2regmask[Op_VecX] = regmask_for_ideal_register(Op_VecX, ret);
   idealreg2regmask[Op_VecY] = regmask_for_ideal_register(Op_VecY, ret);
   idealreg2regmask[Op_VecZ] = regmask_for_ideal_register(Op_VecZ, ret);
+  idealreg2regmask[Op_VecZ] = regmask_for_ideal_register(Op_VecZ, ret);
+  if (Matcher::has_predicated_vectors()) {
+    idealreg2regmask[Op_RegVMask] = regmask_for_ideal_register(Op_RegVMask, ret);
+  } else {
+    idealreg2regmask[Op_RegVMask] = idealreg2regmask[Op_RegI];
```

Comment:
No new register pressure threshold has been introduced for predicate registers, hence just a default initialization for targets which do not support them, since its getting used in raise/lower_pressure 

> Hi Jatin,
> 
> Thanks for the work. Unfortunately, it seems that there are some conflicts (e.g. type definition) with our work at [openjdk/panama-vector#40](https://github.com/openjdk/panama-vector/pull/40). Could you please help to take a look at [openjdk/panama-vector#40](https://github.com/openjdk/panama-vector/pull/40), so that we could improve that as well and reduce the potential merge effort?
> 
> Thanks,
> Ningsheng

Hi Ningsheng,

This is the base version, changes are generic in nature and it also cleans up existing hard coded opmask register usages in X86 port.  Regarding code level naming confits, we can work over them once we land the base patch. I have taken care of your suggestions and comments.

Best Regards,
Jatin

Reply:
> How `idealreg2regmask[Op_RegVMask]` is used when `Matcher::has_predicated_vectors() == false`?

No new register pressure threshold has been introduced for predicate registers, hence just a default initialization for targets which do not support them, since its getting used in raise/lower_pressure